core/src/main/java/org/mule/runtime/core/internal/streaming/FunkyCursorManager.java

```diff
@@ -0,0 +1,239 @@
+    //
+    //if (state != null && state.removeCursor(handle.getCursorProvider(), cursor)) {
+    //  registry.invalidate(eventId);
+    //}
+  }
+
+  private void terminated(StreamingSessionKey key) {
+    registry.invalidate(key);
+    //EventStreamingState state = registry.getIfPresent(key);
+    //if (state != null) {
+    //  registry.invalidate(rootContext.getId());
+    //}
+  }
+
+  private void hookEventTermination(StreamingSessionKey key) {
+    key.getEventContext().onTerminated((response, throwable) -> terminated(key));
+  }
+
+  private class EventStreamingState {
+
+    private final AtomicBoolean disposed = new AtomicBoolean(false);
+    private final Map<Integer, WeakReference<ManagedCursorProvider>> providers = new ConcurrentHashMap<>();
+
+    //private final AtomicInteger cursorCount = new AtomicInteger(0);
+
+    //private final LoadingCache<CursorProvider<CursorStream>, Set<Cursor>> cursors = Caffeine.newBuilder()
+    //    .removalListener((CursorProvider<CursorStream> key, Set<Cursor> value, RemovalCause cause) -> {
+    //      try {
+    //        closeProvider(key);
+    //        releaseAll(value);
+    //      } finally {
+    //        key.releaseResources();
+    //      }
+    //    })
+    //    .build(key -> {
+    //      statistics.incrementOpenProviders();
+    //      return newSetFromMap(new ConcurrentHashMap<>());
+    //    });
+
+    private void addProvider(ManagedCursorProvider provider) {
+      streamingGarbageCollector.track(provider);
+      providers.put(identityHashCode(provider), new WeakReference<>(provider));
+    }
+
+    //private void addCursor(CursorProvider provider, Cursor cursor) {
+    //  cursors.get(provider).add(cursor);
+    //  cursorCount.incrementAndGet();
+    //}
+
+    //private boolean removeCursor(CursorProvider provider, Cursor cursor) {
+    //  Set<Cursor> openCursors = cursors.get(provider);
+    //  if (openCursors.remove(cursor)) {
+    //    statistics.decrementOpenCursors();
+    //    if (cursorCount.decrementAndGet() <= 0 && provider.isClosed()) {
+    //      dispose();
+    //      return true;
+    //    }
+    //  }
+    //
+    //  return false;
+    //}
+
+    private void dispose() {
+      if (disposed.compareAndSet(false, true)) {
+        providers.values().forEach(weakReference -> {
+          ManagedCursorProvider provider = weakReference.get();
+          if (provider != null) {
+            weakReference.clear();
+            provider.releaseResources();
+          }
+        });
+      }
+    }
+    //
+    //private void releaseAll(Collection<Cursor> cursors) {
+    //  cursors.forEach(cursor -> {
+    //    try {
+    //      cursor.release();
+    //      statistics.decrementOpenCursors();
+    //    } catch (Exception e) {
+    //      LOGGER.warn("Exception was found trying to close cursor. Execution will continue", e);
+    //    }
+    //  });
+    //}
+    //
+    //private void closeProvider(CursorProvider provider) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7643&issues=AWirZZoRotIXnmsUDp_s&open=AWirZZoRotIXnmsUDp_s)

[ISSUE_KEY]: AWirZZoRotIXnmsUDp_s